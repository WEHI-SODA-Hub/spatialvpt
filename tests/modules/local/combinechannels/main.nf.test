nextflow_process {

    name "Test Process COMBINECHANNELS"
    script "modules/local/combinechannels/main.nf"
    process "COMBINECHANNELS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'small_set' ],
                    [ file('${projectDir}/data/202305010900_U2OS_small_set_VMSC00000/region_0/images', checkIfExists: true) ],
                ])
                input[1] = "cellbound1,cellbound2,cellbound3"
                input[2] = "mosaic_Cellbound_z"
                input[3] = 4
                input[4] = 512
                input[5] = 0.108
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
