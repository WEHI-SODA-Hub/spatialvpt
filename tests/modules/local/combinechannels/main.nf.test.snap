{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "small_set"
                        },
                        "mosaic_Cellbound_z4.tif:md5,bbbc98eaa4aacfd5dcd445538bf89451"
                    ]
                ],
                "1": [
                    "versions.yml:md5,56806e41fd0c07678b7a463330b4296b"
                ],
                "tif": [
                    [
                        {
                            "id": "small_set"
                        },
                        "mosaic_Cellbound_z4.tif:md5,bbbc98eaa4aacfd5dcd445538bf89451"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,56806e41fd0c07678b7a463330b4296b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-09-03T09:24:12.896801299"
    }
}